collect_app/src/main/java/org/odk/collect/android/formentry/FormEntryViewModel.java

```diff
@@ -125,7 +127,12 @@ public boolean canAddRepeat() {
         }
     }
 
-    public void moveForward() {
+    public void moveForward(FormController formController) {
+        if (this.formController == null) {
```

Comment:
I'm thinking instead of passing `FormController` in here we could just grab the singleton and then wrap that in an `requireFormController` method. That way we're not having to change the signature so hopefully not leaving anything around we'll want to take out later.

In my head, if this all worked nicely the structure would actually be pretty similar anyway:  the instance of `FormController` would be grabbed on demand from a cache anyway (but keyed/bucketed by the form entry session rather than an application singleton) so I could see us ending up with a structure where any method that needs it in the `ViewModel` just loads it in lazily.

I'll have a go at adding that in and push to this, and then we can discuss later.